src/test/java/com/birdie/kafka/connect/smt/DebeziumJsonDeserializerTest.java

```diff
@@ -400,4 +404,86 @@ public void skipsTombstones() {
 
         assertEquals(record, transformedRecord);
     }
+
+    @Test
+    public void unionSchemasAcrossMultipleMessages() {
+        Struct firstMessageContents = new Struct(simpleSchema);
+        firstMessageContents.put("id", "1234-5678");
+        firstMessageContents.put("json", "{\"foo\": \"da value\"}");
+
+        Struct secondMessageContents = new Struct(simpleSchema);
+        secondMessageContents.put("id", "1234-5678");
+        secondMessageContents.put("json", "{\"bar\": \"oh a value\"}");
+
+        Struct thirdMessageContents = new Struct(simpleSchema);
+        thirdMessageContents.put("id", "1234-5678");
+        thirdMessageContents.put("json", "{\"foo\": \"way\", \"bar\": \"plop\"}");
+
+        DebeziumJsonDeserializer transformer = new DebeziumJsonDeserializer();
+        transformer.configure(new HashMap<>() {{
+            put("optional-struct-fields", "true");
+            put("union-previous-messages-schema", "true");
+        }});
+
+        SourceRecord firstTransformed = transformer.apply(sourceRecordFromValue(firstMessageContents));
+        SourceRecord secondTransformed = transformer.apply(sourceRecordFromValue(secondMessageContents));
+        SourceRecord thirdTransformed = transformer.apply(sourceRecordFromValue(thirdMessageContents));
+
+        assertNotNull(firstTransformed.valueSchema().field("json").schema().field("foo"));
+        assertNull(firstTransformed.valueSchema().field("json").schema().field("bar"));
+
+        assertNotNull(secondTransformed.valueSchema().field("json").schema().field("foo"));
+        assertNotNull(secondTransformed.valueSchema().field("json").schema().field("bar"));
+
+        assertEqualsSchemas(secondTransformed.valueSchema(), thirdTransformed.valueSchema());
+    }
+
+    @Test
+    public void unionSchemasAcrossMultipleIncompatibleMessages() {
+        Struct firstMessageContents = new Struct(simpleSchema);
+        firstMessageContents.put("id", "1234-5678");
+        firstMessageContents.put("json", "{\"foo\": \"da value\", \"bar\": \"somethingElse\"}");
```

Comment:
I don't see any tests for object data types vs int/string/etc?
Is it likely to happen?

For example:
`firstMessageContents.put("json", "{\"foo\": \"da value\", \"bar\": \"somethingElse\"}");`
`secondMessageContents.put("json", "{\"foo\": \"da value\", \"bar\": {\"foobar\": \"someval\"}}");`

Reply:
Very true, very true. It would be the exact same scenario as what is tested in this test though, the fact it's _different_ is important, whether it's a list or an object doesn't really matter but happy to add one.

Reply:
What about the other way around, if it starts as an object or list and then gets a string value, is the union able to cope with that? Sorry, not read all the schema details about how it stores or creates them.

I think it's worth adding one to ensure consistency with changes in the future.